hibernate-core/src/main/java/org/hibernate/tool/schema/internal/AbstractSchemaMigrator.java

```diff
@@ -139,6 +135,8 @@ public void doMigration(Metadata metadata, ExecutionOptions options, TargetDescr
 		}
 	}
 
+	protected abstract DatabaseInformation getDatabaseInformation(DdlTransactionIsolator ddlTransactionIsolator, Namespace namespace);
```

Comment:
Rather than introducing a non-compatible abstract method, is it possible to add caching mechanism in `Helper.buildDatabaseInformation()` internally? Seems more straightforward and safe.